packages/Popover/src/Popover.js

```diff
@@ -447,7 +448,13 @@ class Popover extends React.Component {
     return (
       <ThemeContext.Provider value={isDark}>
         <PopoverContext.Provider value={contextValue}>
-          <sc.Popover data-pka-anchor="popover" {...moreProps} ref={this.$popover}>
+          <sc.Popover
+            data-pka-anchor="popover"
+            {...moreProps}
+            ref={this.$popover}
+            onKeyUp={this.handleKeyUp}
+            onKeyDown={this.handleKeyDown}
```

Comment:
What if the key listeners are put on the Popover.Content (via context)? Could you avoid all the DOM checking and just _know_ the event was for _this_ `<Popover>` set `isOpen=false` and stop the propagation?